api/tests/certification/course/integration/infrastructure/repositories/sco-certification-candidate-repository_test.js

```diff
@@ -169,14 +248,73 @@ describe('Certification | Course | Integration | Repository | SCOCertificationCa
       };
       const sessionPublishedId = databaseBuilder.factory.buildSession({ publishedAt: '2024-01-01' }).id;
       const unpublishedSessionId = databaseBuilder.factory.buildSession({ publishedAt: null }).id;
-      databaseBuilder.factory.buildCertificationCandidate({
+      const candidateFromUnpublishedSession = databaseBuilder.factory.buildCertificationCandidate({
         ...candidate,
         sessionId: unpublishedSessionId,
-      }).id;
-      const candidateIdFromPublishedSession = databaseBuilder.factory.buildCertificationCandidate({
+      });
+      const candidateFromPublishedSession = databaseBuilder.factory.buildCertificationCandidate({
         ...candidate,
         sessionId: sessionPublishedId,
-      }).id;
+      });
+      databaseBuilder.factory.buildCertificationCourse({
+        sessionId: unpublishedSessionId,
+        lastName: candidateFromUnpublishedSession.lastName,
+        firstName: candidateFromUnpublishedSession.firstName,
+        isPublished: false,
+        userId: candidateFromUnpublishedSession.userId,
+      });
+      databaseBuilder.factory.buildCertificationCourse({
+        sessionId: sessionPublishedId,
+        lastName: candidateFromPublishedSession.lastName,
+        firstName: candidateFromPublishedSession.firstName,
+        isPublished: true,
+        userId: candidateFromPublishedSession.userId,
+      });
+
+      await databaseBuilder.commit();
+
+      // when
+      const candidatesIds = await scoCertificationCandidateRepository.findIdsByOrganizationIdAndDivision({
+        organizationId,
+        division,
+      });
+
+      // then
+      expect(candidatesIds).to.deep.equal([candidateFromPublishedSession.id]);
+    });
+
+    it('should not retrieve candidates who did not enter the session', async function () {
+      // given
+      const division = '3ème A';
+      const organizationId = databaseBuilder.factory.buildOrganization().id;
+      const candidate = {
+        firstName: 'Smith',
+        lastName: 'Aaron',
+        organizationLearnerId: databaseBuilder.factory.buildOrganizationLearner({
+          organizationId: organizationId,
+          division,
+        }).id,
+      };
+      const sessionIdOne = databaseBuilder.factory.buildSession({ publishedAt: '2024-02-01' }).id;
+      const sessionIdTwo = databaseBuilder.factory.buildSession({ publishedAt: '2024-01-01' }).id;
+      // That candidate has no related certification-course
```

Comment:
```suggestion
      // This candidate has no related certification-course
```
Nitpicking de nitpicking.
Si c'est celui juste après, this c'est bien :)

Reply:
👁️ 
Done